.gitignore

```diff
@@ -28,6 +28,7 @@
 
 # Nested build directory
 /build*
+/sycl/test-e2e/build*
```

Comment:
This change is not related to `__nvvm_reflect_ocl` built-in. Please, move to a separate PR.

I don't understand why we need this change.

Reply:
`.gitignore` change removed

Reply:
This change does not change any behaviours of libclc but it is needed for "upstreamability", since it's better practice to have a libclc written in just openCL instead of needing to mix LLVM IR and openCL which compile into the same bc libs.